src/cookies.js

```diff
@@ -83,27 +83,57 @@ function tryGetDocumentCookieNoInline(win) {
  *     - highestAvailableDomain: If true, set the cookie at the widest domain
  *       scope allowed by the browser. E.g. on example.com if we are currently
  *       on www.example.com.
- *     - domain: Explicit domain to set.
+ *     - domain: Explicit domain to set. domain overrides HigestAvailableDomain
  *     - allowOnProxyOrigin: Allow setting a cookie on the AMP Cache.
  */
 export function setCookie(win, name, value, expirationTime, opt_options) {
   checkOriginForSettingCookie(win, opt_options, name);
-  if (opt_options && opt_options.highestAvailableDomain) {
+  let domain = undefined;
+  // Respect explicitily set domain over higestAvailabeDomain
+  if (opt_options && opt_options.domain) {
+    domain = opt_options.domain;
+  } else if (opt_options && opt_options.highestAvailableDomain) {
+    domain = getHighestAvailableDomain(win);
+  }
+  trySetCookie(win, name, value, expirationTime, domain);
+}
+
+/**
+ * Attemp to find the HighestAvailableDomain on
+ * @param {!Window} win
+ */
+export function getHighestAvailableDomain(win) {
+  // TODO(zhouyx@): Once we decide to allow publisher to put
```

Comment:
we can do it now. the `<meta>` is invalid AMP anyway. once we want to support that, we just need to add a validation rule.

Reply:
I'd prefer to keep the same behavior here and add the support in separate PR after we validate the `<meta name='amp-cookie-scope'>`

Reply:
I feel that is an unnecessary extra step.

Reply:
done